BrowserKit/Sources/SummarizeKit/ToSBottomSheetViewController.swift

```diff
@@ -0,0 +1,198 @@
+        allowButtonA11yLabel: String,
+        cancelButtonTitle: String,
+        cancelButtonA11yId: String,
+        cancelButtonA11yLabel: String,
+        onRequestOpenURL: ((URL?) -> Void)?,
+        onAllowButtonPressed: (() -> Void)?,
+        onDismiss: (() -> Void)?
+    ) {
+        self.titleLabel = titleLabel
+        self.descriptionLabel = descriptionLabel
+        self.linkButtonLabel = linkButtonLabel
+        self.linkButtonURL = linkButtonURL
+        self.onAllowButtonPressed = onAllowButtonPressed
+        self.onRequestOpenURL = onRequestOpenURL
+        self.allowButtonTitle = allowButtonTitle
+        self.allowButtonA11yId = allowButtonA11yId
+        self.allowButtonA11yLabel = allowButtonA11yLabel
+        self.cancelButtonTitle = cancelButtonTitle
+        self.cancelButtonA11yId = cancelButtonA11yId
+        self.cancelButtonA11yLabel = cancelButtonA11yLabel
+        self.onDismiss = onDismiss
+    }
+}
+
+public class ToSBottomSheetViewController: UIViewController,
+                                           UITextViewDelegate,
+                                           BottomSheetChild,
+                                           Themeable {
+    private struct UX {
+        static let contentHorizontalPadding: CGFloat = 29.0
+        static let titleLabelTopPadding: CGFloat = 30.0
+        static let descriptionLabelTopPadding: CGFloat = 20.0
+        static let allowButtonTopPadding: CGFloat = 20.0
+        static let cancelButtonTopPadding: CGFloat = 10.0
+    }
+    public var themeManager: any Common.ThemeManager
+    public var themeObserver: (any NSObjectProtocol)?
+    public var notificationCenter: any Common.NotificationProtocol
+
+    public var currentWindowUUID: Common.WindowUUID?
+    private let viewModel: ToSBottomSheetViewModel
+    public weak var dismissDelegate: BottomSheetDismissProtocol?
+
+    private let titleLabel: UILabel = .build {
+        $0.textAlignment = .center
+        $0.font = FXFontStyles.Bold.headline.scaledFont()
+    }
+    private let descriptionLabel: UITextView = .build {
+        $0.textAlignment = .center
+        $0.isEditable = false
+        $0.isScrollEnabled = false
+    }
+    private let allowButton: PrimaryRoundedButton = .build()
+    private let cancelButton: SecondaryRoundedButton = .build()
+
+    public init(
+        viewModel: ToSBottomSheetViewModel,
+        themeManager: any Common.ThemeManager = AppContainer.shared.resolve(),
+        notificationCenter: any Common.NotificationProtocol = NotificationCenter.default,
+        windowUUID: WindowUUID
+    ) {
+        self.viewModel = viewModel
+        self.themeManager = themeManager
+        self.notificationCenter = notificationCenter
+        self.currentWindowUUID = windowUUID
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override public func viewDidLoad() {
```

Comment:
Could we split the code in this method up into `configure` and `setupLayout`? That would fit more with what we have in other places of the app.